### PR TITLE
Add method to get type by its name

### PR DESCRIPTION
**Pull request by gavrilikhin-d**
Added `context.get_type("name")` method

## Description

Added new method with documentation and tests to context. It uses `LLVMGetTypeByName2` from llvm_sys, which was missing in inkwell.

## Related Issue

Related to my question: #367 

## How This Has Been Tested

Test when type is found and when not found.

## Checklist

- [x] I have read the [Contributing Guide](https://github.com/TheDan64/inkwell/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by gavrilikhin-d** [comment]
How to test compilation with every llvm version?

**Comment by TheDan64** [comment]
Unfortunately you'd need to have that llvm version installed to run it. You can test against the CI build, though...

File: src/context.rs [hunk 1]
```diff
@@ -926,6 +961,24 @@ impl Context {
...
+    ///	let name = "opaque";
+    ///	let opaque = context.opaque_struct_type(name);
+    ///
+    /// let got = context.get_type(name);
+    ///	assert_eq!(got, Some(AnyTypeEnum::from(opaque)));
+    ///
+    /// assert_eq!(context.get_type("non-existent"), None);
+    /// ```
+    #[inline]
+    pub fn get_type<'ctx>(&self, name: &str) -> Option<AnyTypeEnum<'ctx>> {
```
Comment (TheDan64) [inline src/context.rs:978]: We should add this to ContextRef as well

Comment (gavrilikhin-d) [reply src/context.rs:954]: Done in 48af9c33817a881eaa797f954bbf02b2f258c9ba

File: src/context.rs [hunk 2]
```diff
@@ -262,6 +262,17 @@ impl ContextImpl {
         unsafe { StructType::new(LLVMStructCreateNamed(self.0, c_string.as_ptr())) }
     }
```
Comment (gavrilikhin-d) [inline src/context.rs:264]: There is a `LLVMGetTypeByName` function befor 12.0. Should I add support for it too?

**Comment by gavrilikhin-d** [comment]
There is `LLVMGetTypeByName` function befor `llvm12-0`. Should I support it too?

**Comment by gavrilikhin-d** [comment]
Also, is there a reason why inkwell supports such outdated llvm versions?

**Comment by TheDan64** [comment]
If the api is similar enough then yeah, I suppose we can support both

**Comment by gavrilikhin-d** [comment]
`AnyTypeEnum` in `context.rs` is available only with llvm higher than 12.0

**Comment by TheDan64** [comment]
> AnyTypeEnum in context.rs is available only with llvm higher than 12.0

The use import? The enum exists in older versions, so you could extend it

**Comment by gavrilikhin-d** [comment]
Lol, while looking through code for `Module`, i found `get_struct_type(&self, name: &str) -> Option<StructType<'ctx>>` which does exactly the same

**Comment by gavrilikhin-d** [comment]
I think, adding this function to `Context` after `llvm12-0` would be good, though. But it needs to be renamed to `get_struct_type` too.

**Comment by gavrilikhin-d** [comment]
Are there really only struct types, that can be get by name?

**Comment by TheDan64** [comment]
I'm not sure. I think I recall trying it with other types and it only seemed to work for structs for some reason

**Comment by TheDan64** [comment]
But feel free to write some tests and change it if you can repro non structs

**Comment by gavrilikhin-d** [comment]
I found thatLLVM itself assumes that only structs are named types
```cpp
LLVMTypeRef LLVMGetTypeByName2(LLVMContextRef C, const char *Name) {
    return wrap(StructType::getTypeByName(*unwrap(C), Name));
}
```

**Comment by gavrilikhin-d** [comment]
I tried with `"i8"` and other not-struct types - they may not be found by name

**Review by TheDan64 — APPROVED** [review]
Thanks!

**Pull request merged**